### ISSUE TITLE
neovim not registering RPC commands for Node.js remote plugins

### ISSUE THREAD

**Issue by Yilin-Yang** [neovim/node-client#103]
Howdy!

It seems like neovim isn't correctly registering RPC commands for Node.js remote plugins: it detects that the remote plugin exists, but doesn't actually add dictionary entries for the plugin's commands/functions/autocmds in the remote plugin manifest.

I've tried "hardcoding" the remote plugin manifest with the correct values, and once I do that, the remote plugin seems to work fine.

This might be the same issue as described in #53 and/or #43, and possibly fixed by #54, but the issue still occurs on my machine for whatever reason.

## Steps to Reproduce

```bash
cd temp
git clone https://github.com/neovim/node-client.git && cd node-client

cd examples
cat << EOF > .vimrc
set runtimepath+=.
EOF

cd rplugin/node/demo_beautify_css
yarn install

cd ../../..

nvim -u .vimrc +UpdateRemotePlugins
# then exit nvim

cat /home/$USER/.local/share/nvim/rplugin.vim
```

## Expected Output

```vim
" node plugins
call remote#host#RegisterPlugin('node', './rplugin/node/demo_beautify_css', [
      \ {'sync': v:false, 'name': 'BeautifyCSS', 'type': 'command', 'opts': {'nargs': '0'}},
     \ ])


" python3 plugins


" ruby plugins


" python plugins

```

## Actual Output

```vim

" node plugins
call remote#host#RegisterPlugin('node', './rplugin/node/demo_beautify_css', [
     \ ])


" python3 plugins


" ruby plugins


" python plugins

```

## Version Info

Tested on Ubuntu 16.04.

```bash
NVIM v0.3.1
Build type: RelWithDebInfo
LuaJIT 2.0.4
Compilation: /usr/bin/x86_64-linux-gnu-gcc -g -O2 -fPIE -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -DDISABLE_LOG -Wdate-time -D_FORTIFY_SOURCE=2 -Wconversion -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1 -O2 -g -DMIN_LOG_LEVEL=3 -Og -g -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wvla -fstack-protector-strong -fdiagnostics-color=auto -Wno-array-bounds -DINCLUDE_GENERATED_DECLARATIONS -D_GNU_SOURCE -DNVIM_UNIBI_HAS_VAR_FROM -I/build/neovim-xZey2V/neovim-0.3.1/build/config -I/build/neovim-xZey2V/neovim-0.3.1/src -I/usr/include -I/build/neovim-xZey2V/neovim-0.3.1/build/src/nvim/auto -I/build/neovim-xZey2V/neovim-0.3.1/build/include
Compiled by team+vim@tracker.debian.org

Features: +acl +iconv +jemalloc +tui
See ":help feature-compile"

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/usr/share/nvim"

Run :checkhealth for more info
```

```bash
# node.js
v11.4.0
```

## `checkhealth` Output

```markdown

health#nvim#check
========================================================================
## Configuration
  - OK: no issues found

## Performance
  - OK: Build type: RelWithDebInfo

## Remote Plugins
  - OK: Up to date

## terminal
  - INFO: key_backspace (kbs) terminfo entry: key_backspace=\177
  - INFO: key_dc (kdch1) terminfo entry: key_dc=\E[3~

## tmux
  - OK: escape-time: 1ms
  - INFO: $TERM: xterm-256color
  - INFO: default-terminal: tmux
  - ERROR: $TERM differs from the tmux `default-terminal` setting. Colors might look wrong.
    - ADVICE:
      - $TERM may have been set by some rc (.bashrc, .zshrc, ...).

health#provider#check
========================================================================
## Clipboard (optional)
  - OK: Clipboard tool found: xclip

## Python 2 provider (optional)
  - INFO: `g:python_host_prog` is not set.  Searching for python2 in the environment.
  - INFO: Executable: /usr/bin/python2
  - INFO: Python2 version: 2.7.12
  - INFO: python2-neovim version: 0.3.1
  - OK: Latest python2-neovim is installed: 0.3.1

## Python 3 provider (optional)
  - INFO: `g:python3_host_prog` is not set.  Searching for python3 in the environment.
  - INFO: Executable: /usr/bin/python3
  - INFO: Python3 version: 3.5.2
  - INFO: python3-neovim version: 0.3.1
  - OK: Latest python3-neovim is installed: 0.3.1

## Ruby provider (optional)
  - INFO: Ruby: ruby 2.3.1p112 (2016-04-26) [x86_64-linux-gnu]
  - WARNING: `neovim-ruby-host` not found.
    - ADVICE:
      - Run `gem install neovim` to ensure the neovim RubyGem is installed.
      - Run `gem environment` to ensure the gem bin directory is in $PATH.
      - If you are using rvm/rbenv/chruby, try "rehashing".
      - See :help |g:ruby_host_prog| for non-standard gem installations.

## Node.js provider (optional)
  - INFO: Node.js: v11.4.0
  - INFO: Neovim node.js host: /usr/lib/node_modules/neovim/bin/cli.js
  - OK: Latest "neovim" npm/yarn package is installed: 4.2.1

```
Thanks in advance!

**Comment by Yilin-Yang** [comment]
An update:

So I investigated a bit further, and it looks like this issue *only* occurs when specifying `runtimepath` using relative filepaths, i.e. with `.` or `..`.  Amending the instructions above from:

```bash
cd examples
cat << EOF > .vimrc
set runtimepath+=.
EOF
```

To:
```bash
cd examples
cat << EOF > .vimrc
set runtimepath+=/home/yiliny/plugin/node-client/examples
" or,
execute 'set runtimepath+=' . getcwd()
EOF
```

Produces the expected output of:
```vim
call remote#host#RegisterPlugin('node', '/home/yiliny/plugin/node-client/examples/rplugin/node/demo_beautify_css', [
      \ {'sync': v:false, 'name': 'BeautifyCSS', 'type': 'command', 'opts': {}},
     \ ])
```

tl;dr my bad, the original report was more of a [PEBCAK](https://en.wiktionary.org/wiki/PEBCAK) and actually isn't (at all) an issue with `node-client` itself.

Great work with the client! I've been using it pretty heavily for my next plugin project, and it's been a joy to use (leaving aside my own goofs 😅)!
